pil-analyzer/src/type_inference.rs

```diff
@@ -220,15 +260,41 @@ impl TypeChecker {
                                 format_type_scheme_around_name(name, &Some(builtin.clone()))
                             );
                         };
-                        builtin.clone()
+                        DeclaredType {
+                            source,
+                            vars: builtin.vars.clone(),
+                            ty: TypeDeclaredType::Type(builtin.ty.clone()),
+                        }
                     }
                     // Store an (uninstantiated) type scheme for symbols with a declared polymorphic type.
-                    (None, Some(type_scheme)) => type_scheme.clone(),
+                    (None, Some(type_scheme)) => {
+                        let ty = match self.struct_declarations.get(name) {
```

Comment:
I'm not 100% sure about this, and my plan was to get rid of structs_declarations. But, since I don't have the structs expressions, I need some way to match them.

Reply:
Shouldn't the struct declaration be inside `value`?

Reply:
Ah no it's not. But then we just have to pull the DeclaredType up into pil_analyzer and construct it there.